lib/trino-parquet/src/test/java/io/trino/parquet/reader/TestColumnReader.java

```diff
@@ -20,13 +20,19 @@
 import io.trino.parquet.PrimitiveField;
 import io.trino.parquet.RichColumnDescriptor;
 import io.trino.spi.block.Block;
+import org.apache.parquet.bytes.ByteBufferInputStream;
```

Comment:
Please add tests to `AbstractTestParquetReader` instead of this class.
To make the parquet-mr writer produce files with this encoding we need the ability to turn on this encoding through JobConf at `ParquetTester#assertRoundTrip` similar to the way dictionary encoding is enabled. This is needed because `enableByteStreamSplit` is turned off by default in ParquetProperties. For this you'll probably have to propose a PR in parquet-mr for adding the config to JobConf. Or you can try using `org.apache.parquet.hadoop.ParquetWriter` to write files as that allows enabling this encoding.
Another possibility is to implement this encoding in the optimised parquet writer in Trino. However, we would need to have compatibility tests with Hive and Spark (TestHiveCompatibility, TestHiveSparkCompatibility) to verify that this encoding can be successfully read there. We would also need a config to allow turning it off as older versions of Trino and other engines would not be able to read this encoding.

Reply:
@raunaqmorarka Thanks for the comments.  Looking at all the options I think using `org.apache.parquet.hadoop.ParquetWriter` would be easiest. If I write a test case which plainly writes Double/Float values using `org.apache.parquet.hadoop.ParquetWriter` and read it using `org.apache.parquet.hadoop.ParquetReader` and validate the values would that be sufficient ?  `ParquetTester#assertRoundTrip` seems to be supporting lot of options and varied inputs. Please let me know.

Reply:
You can write a separate test similar to `TestTimestamp` which uses `org.apache.parquet.hadoop.ParquetWriter` instead of ParquetTester to write a file using BYTE_STREAM_SPLIT encoding for float and doubles. It needs to verify reads by creating a reader through `FileFormat#createFileFormatReader` (see `TestTimestamp#testReadingAs`)

Reply:
@raunaqmorarka  could you please review again ? Thanks !